backend/core/src/migration/1658992843452-add-afs-related-properties-to-listing.ts

```diff
@@ -0,0 +1,77 @@
+import { Rule } from "src/application-flagged-sets/types/rule-enum"
+import { MigrationInterface, QueryRunner } from "typeorm"
+
+export class addAfsRelatedPropertiesToListing1658992843452 implements MigrationInterface {
+  name = "addAfsRelatedPropertiesToListing1658992843452"
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `ALTER TABLE "listings" ADD "afs_last_run_at" TIMESTAMP WITH TIME ZONE DEFAULT '1970-01-01'`
+    )
+    await queryRunner.query(
+      `ALTER TABLE "listings" ADD "last_application_update_at" TIMESTAMP WITH TIME ZONE DEFAULT '1970-01-01'`
+    )
+    await queryRunner.query(
+      `ALTER TABLE "application_flagged_set" ADD "rule_key" character varying NOT NULL`
+    )
+    await queryRunner.query(
+      `ALTER TABLE "application_flagged_set" ADD CONSTRAINT "UQ_2983d3205a16bfae28323d021ea" UNIQUE ("rule_key")`
+    )
+
+    // set rule_key for existing afses
+    const afsas = await queryRunner.query(`
+      SELECT application_flagged_set_id, applications_id
+      FROM application_flagged_set_applications_applications
+    `)
+    for (const afsa of afsas) {
+      const afs = await queryRunner.query(
+        `
+        SELECT id, listing_id, rule, rule_key
```

Comment:
could this become part of the above query as an inner join so we can cut down on how many times we connect to the db